File: Blog/Blog/src/main/java/com/example/Blog/model/Blog.java
package com.example.Blog.model;
import jakarta.persistence.*;

import lombok.*;

import java.util.Date;
import java.util.List;


@Data
@Entity
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Blog {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;
    private String Title;
    private Date date;
    private String description;
    private String country;
    private String address;
    @Lob
    @Column(length = 1048576)
    private Byte[] pictures;
    @ElementCollection
    @CollectionTable(name = "blog_hashtags")
    private List<String> hashtags;
    /*
    @ManyToOne
    @JoinColumn(name = "user_id")
    private ApplicationUser user;
    */
}

File: TemplateFinal/src/app/models/Blog.ts
export class Blog {
    id!: number;
    iduser!: number;
    title!: string;
    date!: Date;
    description!: string;
    country!: string;
    address!: string;
    comments_id!: number;
    reactions_id!: number[];
    hashtags!: string[];
    pictures!: string; // assuming base64 encoded image string
}
